fluent-plugin-protobuf/lib/fluent/plugin/parse_protobuf.rb

```diff
@@ -13,16 +13,14 @@ class ProtobufParse < Fluent::Plugin::Parser
       def parse(text)
         begin
           inflated = Snappy.inflate(text)
-          log.info("HIT PARSER: #{Base64.encode64(inflated)}")
-  
           decoded = Prometheus::WriteRequest.decode(inflated)
```

Comment:
One question, do we use official Prometheus proto schemas compiled `WriteRequest` class to decode ?

Reply:
Unfortunately no. Since the `remote.proto` and `types.proto` files are written in `proto3` syntax, but `gogoproto` is written in `proto2`, we cannot compile the files into Ruby currently. According to the [protobuf docs](https://developers.google.com/protocol-buffers/docs/reference/ruby-generated), `For the moment only proto3 is supported. Support for proto2 is planned, but not available yet.`

I tried a hacky thing by changing the `gogo.proto` file to `syntax = "proto3"` (and took out all of the `optional` keywords in the file since that is `proto2` syntax). Then I could compile `remote.proto` and `types.proto` without the compiler complaining that `gogo.proto` was `proto2`. However, the only difference in the generated Ruby file compared to when taking out the `gogoproto` dependency from the `remote.proto` and `types.proto` files was one line:
```
require 'gogoproto/gogo_pb'
```
so I believe the schema we are using here is valid and consistent with the official Prometheus proto schema.

But unfortunately I don't think there is anything we can do about this until the Ruby compiler supports `proto2`, or `gogoproto` is updated to `proto3`.